### DISCUSSION TITLE
Creality CR-10 Smart pro

### DISCUSSION THREAD

**Discussion by Yearra** [OrcaSlicer/OrcaSlicer#3542] [General]
Hi everyone, 

I have a Creality CR-10 Smart Pro and I would like to use Orca Slicer but I do not see a profile for my machine. 
I have also been searching the web but couldn't find a profile for Orca slicer. 

Is there anyone using the same machine with Orca Slicer?

**Comment by Fannar1984** [comment]
i just use the cr10 v2 profile and add gcode but it will be nice if some one have the profile and can share the ini file

**Comment by walt-karr** [comment]
I too run a CR-10 Smart Pro with a Sonic Pad and have been using Cura Slicer. But after seeing much positive feedback from so many 3D Printing YouTubers, thought I would give Orca Slicer a try.  But finding that one of Creality's Flagship printer's profile isn't supported in the Orca Slicer set up has put me off.  I don't have the time to not only learn a new Slicer program, but to customize or custom build a printer profile as well. I will have to pass for now and check back to see if the CR-10 Smart Pro is supported in future updates.  Mean while I will use my hobby time printing things with my current set up.

**Reply by walt-karr** [reply]
Ok, I changed my mind and here is what I'm using for beginning and ending G-code:

Beginning G-code

G90 ; use absolute coordinates
M83 ; extruder relative mode
M104 S[nozzle_temperature_initial_layer] ; set extruder temp
M140 S[bed_temperature_initial_layer_single] ; set bed temp
M190 S[bed_temperature_initial_layer_single] ; wait for bed temp
M109 S[nozzle_temperature_initial_layer] ; wait for extruder temp
G28 ; home all
G29 ;auto bed levelling
G1 Z2 F240
G1 X2 Y10 F3000
G1 Z0.28 F240
G92 E0
G1 Y190 E15 F1500 ; intro line
G1 X2.3 F5000
G92 E0
G1 Y10 E15 F1200 ; intro line
G92 E0

Ending G-code


{if max_layer_z < printable_height}G1 Z{z_offset+min(max_layer_z+2, printable_height)} F600{endif} ; Move print head up
G1 X5 Y170 F{travel_speed*60} ; present print
{if max_layer_z < printable_height-10}G1 Z{z_offset+min(max_layer_z+70, printable_height-10)} F600{endif} ; Move print head further up
M140 S0 ; turn off heatbed
M104 S0 ; turn off temperature
M107 ; turn off fan
M84 X Y E ; disable motors

**Reply by digital-slave** [reply]
Hello,

G29 will level the bed every time you don't need to do that.  I was actually  looking for the Gcode to ensure bedevilling is implimented..

**Reply by digital-slave** [reply]
Hi if you replace G29 with 
M420 S1 Z10; Apply stored bed leveling mesh linearilly until z-height 10
you can use the last stored bed level that is stored on the cr-10.
